CMakeLists.txt

```diff
@@ -187,31 +187,36 @@ if(WITH_BLUESTORE)
 endif()
 
 include(CMakeDependentOption)
+CMAKE_DEPENDENT_OPTION(WITH_LIBBLK "Enable backend driver to access storage device" ON
+  "WITH_BLUESTORE OR WITH_RBD_RWL" OFF)
+
 CMAKE_DEPENDENT_OPTION(WITH_LIBURING "Enable io_uring bluestore backend" OFF
-  "WITH_BLUESTORE;HAVE_LIBAIO" OFF)
+  "WITH_LIBBLK;HAVE_LIBAIO" OFF)
 set(HAVE_LIBURING ${WITH_LIBURING})
 
 CMAKE_DEPENDENT_OPTION(WITH_SYSTEM_LIBURING "Require and build with system liburing" OFF
-  "HAVE_LIBAIO;WITH_BLUESTORE" OFF)
+  "WITH_LIBBLK;HAVE_LIBAIO" OFF)
 
-CMAKE_DEPENDENT_OPTION(WITH_BLUESTORE_PMEM "Enable PMDK libraries" OFF
-  "WITH_BLUESTORE" OFF)
+CMAKE_DEPENDENT_OPTION(WITH_PMEM "Enable PMDK libraries" OFF
+  "WITH_LIBBLK" OFF)
+if(WITH_PMEM)
+  if(NOT WITH_LIBBLK)
+    message(SEND_ERROR "Please enable LIBBLK for using PMDK")
+  endif()
```

Comment:
i don't think this is necessary. see https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html and https://github.com/Kitware/CMake/blob/master/Modules/CMakeDependentOption.cmake. so if `WITH_LIBBLK` is OFF, `WITH_PMEM` is disabled as well.

Reply:
@tchaikov 
Thanks. I'll remove the redundant check here.